### ISSUE TITLE
Improve SQL rendering to include sorted() operations

### ISSUE THREAD

**Issue by minborg** [speedment/speedment#400]
Stream pipelines like 
``` java
        Stream<Film> stream = films.stream();

        if (ratings != null) {
            stream = stream.filter(Film.RATING.equal(ratings));
        }
        
        return stream
            .sorted(Film.LENGTH.comparator())
            .skip(page * PAGE_SIZE)
            .limit(PAGE_SIZE)
            .collect(toList());
```
can be optimized more.

**Comment by paxbit** [comment]
Is this supposed to be final? It is closed and tagged as released by 3.0.6, which I'm currently using.

Does not seem to work for me regarding limiting result set size by rendering LIMIT (and OFFSET) clauses, at least not when using Postgres. According to com/speedment/runtime/core/internal/component/sql/optimizer/FilterSortedSkipOptimizer.java:204 LIMIT will always be fixed to Long.MAX_VALUE.

The following code:
```
dataManager.stream()
  .filter(somePredicateForSomeField)
  .skip(10L)
  .limit(10L)
  .collect(Collectors.toList())
```

generates this SQL:
`SELECT ... FROM "public"."data" WHERE ("public"."sometable"."somefield" = ?) LIMIT 223372036854775807 OFFSET ?, values:[somevalue, 10]`

The limit does not seem to be properly rendered.

It same applies for:
```
dataManager.stream()
  .sorted(...)
  .skip(10L)
  .limit(10L)
  .collect(Collectors.toList())
```

```
dataManager.stream()
  .skip(10L)
  .limit(10L)
  .collect(Collectors.toList())
```

**Comment by Pyknic** [comment]
We decided to do the update in two steps. The skip was really the critical one to add to the query since it can potentially result in loading thousands of unnecessary rows. The limit is nice to fix, but not critical since the stream can still be terminated prematurely once 10 rows have been loaded, even if they are not represented in the query. 

We still plan on adding the limit-operation to the query in the near future (as described in #413).
